recipe/meta.yaml

```
@@ -43,6 +43,12 @@ requirements:
     - gitpython >=3.0.7,<4,!=3.1.19
     - pydeck >=0.8.0,<1
     - tornado >=6.0.3,<7
+    - importlib-metadata >=1.4,<8
+    - pydeck >=0.8.0b4,<1
+    - python-dateutil >=2.7.3,<3
+    - typing-extensions >=4.3.0,<5
+    - tzlocal >=1.1,<6
+    - validators >=0.2,<1
   run_constrained:
     - snowflake-snowpark-python >=0.9.0 # [py<312]
     - snowflake-connector-python >=2.8.0 # [py<312]
```

```suggestion
    - snowflake-snowpark-python >=1.17.0  # [py<312]
    - snowflake-connector-python >=2.8.0  # [py<312]
```

nb two spaces before the selector

---

Why raise the upper bound?

---

```
EXTRA_REQUIRES = {
    "snowflake": [
        "snowflake-snowpark-python[modin]>=1.17.0; python_version<'3.12'",
        "snowflake-connector-python>=2.8.0; python_version<'3.12'",
    ]
}
```
https://github.com/streamlit/streamlit/blob/1.38.0/lib/setup.py#L74-L75

---

Done